datafusion/core/src/physical_optimizer/pruning.rs

```diff
@@ -1648,22 +1649,46 @@ fn build_like_match(
             let s = unpack_string(lit.value())?;
             return Ok(s);
         }
-        plan_err!("LIKE expression must be a string literal")
+        plan_err!("Unexpected LIKE expression: {expr:?}")
     }
 
-    // I *think* that ILIKE could be handled by making the min lowercase and max uppercase
-    // but that requires building the physical expressions that call lower() and upper()
+    /// Try and increment the the string's bytes from right to left, returning when the result
+    /// is a valid UTF8 string. Returns `None` when it can't increment any byte.
+    /// Vendored from https://github.com/apache/arrow-rs/blob/56525efbd5f37b89d1b56aa51709cab9f81bc89e/parquet/src/column/writer/mod.rs#L1432-L1448
+    fn increment_utf8(data: &str) -> Result<String> {
```

Comment:
define the function outside of `build_like_match` and add unit tests for it.

Reply:
done